### ISSUE TITLE
Rustdoc comments to title/description fields

### ISSUE THREAD

**Issue by davidgraeff** [GREsau/schemars#7]
The clap and structopt crates use the rustdoc comment for describing cli arguments.
I think the same approach can be used to fill jsonschemas title and description fields.

One way of interpreting those is, to use the rustdoc comment for `description`, expect if the first line starts with a markdown heading character ("#") which would be used for the `title` instead. Example:

```rust
struct foo {
   /// # The title
   /// The description text
   an_int: u16
};
```

WDYT?

**Comment by GREsau** [comment]
Sounds like a good idea to me!

I was considering just using the first line of a rustdoc comment for the `title` since that would be a short summary [by convention](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#summary-sentence). But a summary isn't quite the same as a title, so I actually prefer your proposal of checking for a leading `#` character, which makes it clearer that it's a title.

**Comment by GREsau** [comment]
I had some time today to implement this, and it's now published to crates.io as a prerelease version: https://crates.io/crates/schemars/0.6.0-doc

I want to do a bit of cleanup before I publish it as 0.6.0 properly, but the actual functionality is pretty much done if you want to give it a try

**Comment by davidgraeff** [comment]
That's awesome. I didn't come to it yet. I'll give it a try. btw: you can release "prereleases" as well with cargo. They will not be defaulted to when someone adds the crate as dependency. The tokio crate did this for quite a while for users to test.

**Comment by GREsau** [comment]
Implemented in v0.6, now published to crates.io (https://crates.io/crates/schemars/0.6.1)
